File: media/js/firefox/whatsnew/whatsnew-75.js
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

// create namespace
if (typeof window.Mozilla === 'undefined') {
    window.Mozilla = {};
}

(function(Mozilla) {
    'use strict';

    var client = Mozilla.Client;
    var body = document.getElementsByTagName('body')[0];

    // bug 1419573 - only show "Your Firefox is up to date" if it's the latest version.
    if (client.isFirefoxDesktop) {
        client.getFirefoxDetails(function(data) {
            if (data.isUpToDate) {
                document.querySelector('.c-page-header').classList.add('show-up-to-date-message');
            }
        });

        client.getFxaDetails(function(details) {
            body.classList.remove('state-fxa-default');

            // if signed in with more than one device, show the Monitor CTA.
            if ((details.setup && details.browserServices.sync.mobileDevices > 0) || window.location.search.indexOf('has-devices=true') !== -1) {
                body.classList.add('state-fxa-has-devices');
            }
            // else if signed in with no other devices, show connect device CTA.
            else if ((details.setup && details.browserServices.sync.mobileDevices === 0) || window.location.search.indexOf('has-no-devices=true') !== -1) {
                body.classList.add('state-fxa-has-no-devices');
            }
        });
    }

})(window.Mozilla);

File: tests/pages/firefox/whatsnew/whatsnew_75.py
# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at http://mozilla.org/MPL/2.0/.

from selenium.webdriver.common.by import By

from pages.base import BasePage


class FirefoxWhatsNew75Page(BasePage):

    URL_TEMPLATE = '/{locale}/firefox/75.0/whatsnew/all/{params}'

    _signed_out_sync_button_locator = (By.CSS_SELECTOR, '.show-sync .qa-sync-sign-in-button')
    _signed_in_connect_device_button_locator = (By.CSS_SELECTOR, '.show-sync .qa-sync-connect-device-button')
    _signed_in_monitor_button_locator = (By.CSS_SELECTOR, '.show-monitor .qa-monitor-button')

    def wait_for_page_to_load(self):
        self.wait.until(lambda s: self.seed_url in s.current_url)
        el = self.find_element(By.TAG_NAME, 'body')
        self.wait.until(lambda s: 'state-fxa-default' not in el.get_attribute('class'))
        return self

    @property
    def is_signed_out_sync_button_displayed(self):
        return self.is_element_displayed(*self._signed_out_sync_button_locator)

    @property
    def is_signed_in_monitor_button_displayed(self):
        return self.is_element_displayed(*self._signed_in_monitor_button_locator)

    @property
    def is_signed_in_connect_device_button_displayed(self):
        return self.is_element_displayed(*self._signed_in_connect_device_button_locator)
